src/main/kotlin/org/rust/lang/core/types/infer/TypeInferenceWalker.kt

```diff
@@ -1333,7 +1333,7 @@ class RsTypeInferenceWalker(
             name == "format_args" -> items.Arguments.asTy()
             name == "unimplemented" || name == "unreachable" || name == "panic" -> TyNever
             name == "write" || name == "writeln" -> inferredTy
-            else -> TyUnknown
+            else -> inferredTy
```

Comment:
I am wondering if there are other such macros besides `todo!`. Do you know how we can check that we covered them all? 🤔

Reply:
The same bug could apply to other format-like macros ("format", "format_args", "format_args_nl", "write", "writeln", "print", "println", "eprint", "eprintln", "panic", "unimplemented", "unreachable", "todo"), but in this PR a fix the general case, so all of them should be covered. And I have no idea about other possible bugs like this